File: java/src/main/java/com/languageweaver/sdk/samples/translations/text/TextTranslationService.java
package com.languageweaver.sdk.samples.translations.text;

import com.languageweaver.sdk.common.LanguageWeaverClient;
import com.languageweaver.sdk.common.SdkFactory;
import com.languageweaver.sdk.common.configurations.ClientConfiguration;
import com.languageweaver.sdk.common.constants.Format;
import com.languageweaver.sdk.translate.common.request.TranslateTextRequest;
import com.languageweaver.sdk.translate.common.result.TranslateTextResult;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class TextTranslationService {

    public static void main(String[] args) throws Exception {
        try (LanguageWeaverClient lwClient = new SdkFactory().getLanguageWeaverClient(new ClientConfiguration())) {
            translateText(lwClient);
            translateTextWithLinguisticOptions(lwClient);
        }
    }

    private static void translateText(LanguageWeaverClient lwClient) throws IOException, InterruptedException {
        TranslateTextRequest translateTextRequest = new TranslateTextRequest()
                .setSourceLanguageId("eng")
                .setTargetLanguageId("fra")
                .setModel("generic")
                .addInput("The weather is wonderful today!")
                .setInputFormat(Format.PLAIN)
                // provide list of dictionaries
                .addDictionary("dictionaryId or name");
        final TranslateTextResult translateTextResult = lwClient.translateText(translateTextRequest);

        System.out.println(translateTextResult.getTranslation());
    }

    private static void translateTextWithLinguisticOptions(LanguageWeaverClient lwClient) throws IOException, InterruptedException {
        Map<String, String> linguisticOptions = new HashMap<>();
        linguisticOptions.put("Spelling", "US");
        TranslateTextRequest translateTextRequest = new TranslateTextRequest()
                .setSourceLanguageId("fra")
                .setTargetLanguageId("eng")
                .setModel("generic")
                .addInput("j'aime cette couleur")
                .setLinguisticOptions(linguisticOptions)
                .setInputFormat(Format.PLAIN)
                // provide list of dictionaries
                .addDictionary("dictionaryId or name");
        final TranslateTextResult translateTextResult = lwClient.translateText(translateTextRequest);

        System.out.println(translateTextResult.getTranslation());
    }
}

File: javascript/browser/src/samples/translations/text/TextTranslationService.js
import SdkFactory, {
    ClientConfiguration,
    CredentialsConfiguration,
    Format,
    TranslateTextRequest
} from "@language-weaver/lw-sdk-js";

export const translateText = async (clientId, clientSecret) => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId, clientSecret);
        const lwClient = await SdkFactory.getLanguageWeaverClient(clientConfiguration);
        const translateTextRequest = new TranslateTextRequest();
        translateTextRequest.sourceLanguageId = "eng";
        translateTextRequest.targetLanguageId = "fra";
        translateTextRequest.model = "generic";
        // provide list of dictionaries
        // translateTextRequest.addDictionary("dictionaryId or name");
        translateTextRequest.addInput("The weather is wonderful today!");
        translateTextRequest.inputFormat = Format.PLAIN;
        const translateTextResult = await lwClient.translateText(translateTextRequest);

        console.log(translateTextResult.translation);
    } catch (e) {
        console.log(e);
    }
}

export const translateTextWithLinguisticOptions = async (clientId, clientSecret) => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId, clientSecret);
        const lwClient = await SdkFactory.getLanguageWeaverClient(clientConfiguration);
        const translateTextRequest = new TranslateTextRequest();
        translateTextRequest.sourceLanguageId = "fra";
        translateTextRequest.targetLanguageId = "eng";
        translateTextRequest.model = "generic";
        translateTextRequest.addInput("j'aime cette couleur");
        translateTextRequest.inputFormat = Format.PLAIN;

        const linguisticOptions = new Map();
        linguisticOptions.set("Spelling", "UK");
        translateTextRequest.linguisticOptions = linguisticOptions;

        const translateTextResult = await lwClient.translateText(translateTextRequest);

        console.log(translateTextResult.translation);
    } catch (e) {
        console.log(e);
    }
}